src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyOperationManager.cs

```diff
@@ -184,6 +189,24 @@ public virtual bool SetupChannel(IEnumerable<string> sources, CancellationToken
             return true;
         }
 
+        private bool LaunchTestHostAsync(TestProcessStartInfo testHostStartInfo, CancellationToken token)
+        {
+            this.CancellationTokenSource.Token.ThrowIfCancellationRequested();
```

Comment:
yup, got that but most clients would not expect a TestRunComplete when negotiating protocol unless told otherwise.... Was a separate message for cancellation considered for scenarios like these? Might be a bigger change but worth a fluid protocol.